### ISSUE TITLE
Pass status code, headers, from onResourceReceived to page.property('content').then(function(content)

### ISSUE THREAD

**Issue by laurentdebricon** [amir20/phantomjs-node#375]
Hello,
with phantom 1.x, i could set a global variable to pass statusCode and headers from 
onResourceReceived

Then in the callback of page.content, i could handle error 404, or a redirection.

Now with phantom 2.x, It seems impossible. A variable set in onResourceReceived (even a global), is not available in : 

```
page.property('content').then(function(content){

if(GLOBALVARSTATUS == 200){
  res.send(content);
}else if(GLOBALVARSTATUS == 301){
 ....
}
}
```

What is the current solution to handle 404, 301 redirect with phantom node 2.x ?

**Comment by amir20** [comment]
Can you show me the code for 1.x? I don't get how something `GLOBALVARSTATUS` in the callback would have worked because that's executed in node not phantomjs. Maybe seeing how you were doing it in 1.x could help me.

**Comment by laurentdebricon** [comment]
Here is my full code 1.x : 
http://pastebin.com/LbL1UzqD
I am using phantom to render my single page application for Google ... But now that you're telling me it's impossible to share value between onResourceReceived & evaluate , maybe my hack has never worked :s

**Comment by amir20** [comment]
I took a look. I am not sure that ever did work. :) `RESPONSE` is executed in the context of phantomjs and the callback for `evaluate` is executed in the context of nodejs. Those are two different processes. So I doubt it worked. 

Back to the real question, what are you trying to do?

**Comment by laurentdebricon** [comment]
```
PAGE.property('content').then(function(content) {

Here inside I would like to know if it was a HTTP code 200 / 404 / (301 with access to the headers of the request to know to which url it want to be redirected)

});
```

:)

**Comment by amir20** [comment]
By `it` you mean all the resources, right?

**Comment by laurentdebricon** [comment]
The first ressource is enough

**Comment by amir20** [comment]
I did a quick test and I was able to get it working with [this pull request](https://github.com/amir20/phantomjs-node/pull/377/files). 

However, as you can see, I use `eval` to evaluate global space. It is really hacky. I am not sure how I feel about it. 

See the test case at the bottom to see how to use it. I am not convinced that this is something that should be added :(

**Comment by amir20** [comment]
See #378. I added new functionality to get back global space properties. I struggle if I want to document this because it really is a hack and I don't think it makes sense for it to be public.

**Comment by amir20** [comment]
@laurentdebricon any update to this? Does the new pull request work for you?

**Comment by laurentdebricon** [comment]
Hello,
Great ! it's working :


```
page.property('content').then(function(content) {
       ph.windowProperty('lastResponse').then(function(lastResponse) {
                  // here i have access to lastResponse that I set in page.property('onResourceReceived',

      });
});

```

package.json devDependencies misses babel-register. 
npm install babel-register solve the problem, then npm install build the project (it took me some times :p )


Thank you

**Comment by amir20** [comment]
What does Babel register do? I have never used it. 

Sent via my iPhone

**Comment by amir20** [comment]
I released 2.0.1 so you shouldn't need to build locally anymore.

**Comment by laurentdebricon** [comment]
Great ! Thank you for your work !
